### ISSUE TITLE
Hyphenation problems in french

### ISSUE THREAD

**Issue by gnodet** [simoncozens/sile#327]
I spotted at least 2 hyphenation problems in french:
  * a word can't be broken just after an apostrophe (for example `l’homme` should not be hyphenated as `l’-homme`)
  * a word can't be separated from the punctuation (for example `« Soyez` should not be hyphenated as `« -Soyez`, the space being an unbreakable space)

**Comment by simoncozens** [comment]
Hi Guillaume, have you seen this happening in a document? Because SILE doesn't seem to be producing hyphenations for those nodes:

```
> opts = SILE.font.loadDefaults({language="fr"})
> function see(nl) return SU.concat(SU.map(function (x) return x:toText() end, nl),"") end
> see(SILE.hyphenate(SILE.shaper:createNnodes("l’homme",opts)))
"l’homme"
> see(SILE.hyphenate(SILE.shaper:createNnodes("l’homunculus",opts)))
"l’ho_mun_cu_lus"
```

Also, what version are you using? We have been working on Unicode interactions with hyphenation recently, (most recently in 9d6691f) it may be that this was fixed then.

**Comment by gnodet** [comment]
I'm currently using 0.9.3 because #326
And yes, I've seen it in my document output, see the following
<img width="298" alt="hyphen2" src="https://cloud.githubusercontent.com/assets/84022/15924050/898f998e-2e30-11e6-8879-ad676b6ad787.png"> 
<img width="289" alt="hyphen1" src="https://cloud.githubusercontent.com/assets/84022/15924051/89b84582-2e30-11e6-8484-d3bd4b18d0dc.png">

**Comment by simoncozens** [comment]
OK, I'll fix #326 and this should be cleared up. :-)
